sympy/geometry/polygon.py

```diff
@@ -638,6 +642,26 @@ def arbitrary_point(self, parameter='t'):
             perim_fraction_start = perim_fraction_end
         return Piecewise(*sides)
 
+    def parameter_value(self,other):
+        from sympy.geometry.point import Point
+        from sympy.core.symbol import Dummy
+        from sympy.solvers.solvers import solve
+        if not isinstance(other,GeometryEntity):
+            other = Point(other,dim = self.ambient_dimension)
+        if not isinstance(other,Point):
+            raise ValueError("other must be a point")
+        t = Dummy()
+        p = self.arbitrary_point(t)
+        num_conditions = len(p.args)
+        for c in p.args:
+            arb_point = c.args[0]
+            sol = solve(arb_point - other,t,dict = True)
+            if len(sol) != 0:
+                value = sol[0].values()[0]
```

Comment:
This fails in Python 3, where .values() does not return a list. See [Stack Overflow](https://stackoverflow.com/a/16228268) for example. `value = list(sol[0].values())[0]` would work in both Python 2 and Python 3.

Reply:
But I suggest using `value = sol[0].get(t)` which is more concise and works in both: see https://stackoverflow.com/q/11041405